File: softphone-ios/MyLineSoftphone/Models/CallRecord.swift
import Foundation
import GRDB

/// One row in the call history. Mirrors Android `CallRecord.kt` (Room entity).
struct CallRecord: Codable, FetchableRecord, MutablePersistableRecord, Identifiable, Equatable {
    static let databaseTableName = "call_history"

    var id: Int64?
    var direction: CallDirection
    var remoteNumber: String
    var remoteName: String = ""
    var status: String = ""
    var startedAt: Date = Date()
    var answeredAt: Date?
    var endedAt: Date?
    var durationSeconds: Int = 0

    mutating func didInsert(_ inserted: InsertionSuccess) {
        id = inserted.rowID
    }
}

File: softphone-android/app/src/main/java/com/mylinetelecom/softphone/models/CallRecord.kt
package com.mylinetelecom.softphone.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "call_history")
data class CallRecord(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val direction: CallDirection,
    val remoteNumber: String,
    val remoteName: String = "",
    val status: String = "",
    val startedAt: Long = System.currentTimeMillis(),
    val answeredAt: Long? = null,
    val endedAt: Long? = null,
    val durationSeconds: Int = 0
)
